core/domain/email_manager_test.py

```diff
@@ -88,6 +90,198 @@ def test_sender_id_validation(self):
+                    'new_member_role': rights_manager.ROLE_EDITOR,
+                }, csrf_token=csrf_token)
+
+            messages = self.mail_stub.get_sent_messages(to=self.NEW_USER_EMAIL)
+            self.assertEqual(len(messages), 1)
+
+    def test_email_is_not_sent_if_recipient_has_declined_such_emails(self):
+        user_services.update_email_preferences(self.new_user_id, True, False)
+
+        with self.can_send_emails_ctx, self.can_send_editor_role_email_ctx:
+            email_manager.send_role_notification_email(
+                self.editor_id, self.new_user_id, rights_manager.ROLE_OWNER,
+                self.exploration.id, self.exploration.title)
+
+            messages = self.mail_stub.get_sent_messages(to=self.NEW_USER_EMAIL)
+            self.assertEqual(len(messages), 0)
+
+    def test_role_emails_sent_are_correct(self):
+        with self.can_send_emails_ctx, self.can_send_editor_role_email_ctx:
+            email_manager.send_role_notification_email(
+                self.editor_id, self.new_user_id, rights_manager.ROLE_VIEWER,
+                self.exploration.id, self.exploration.title)
+
+            messages = self.mail_stub.get_sent_messages(to=self.NEW_USER_EMAIL)
+            self.assertEqual(len(messages), 1)
+
+            all_models = email_models.SentEmailModel.get_all().fetch()
+            self.assertEqual(len(all_models), 1)
+
+            sent_email_model = all_models[0]
+
+            # Check that email details are correct.
+            self.assertEqual(
+                sent_email_model.recipient_id,
+                self.new_user_id)
+            self.assertEqual(
+                sent_email_model.recipient_email, self.NEW_USER_EMAIL)
+            self.assertEqual(
+                sent_email_model.sender_id, feconf.SYSTEM_COMMITTER_ID)
+            self.assertEqual(
+                sent_email_model.sender_email,
+                'Site Admin <%s>' % feconf.SYSTEM_EMAIL_ADDRESS)
+            self.assertEqual(
+                sent_email_model.intent,
+                feconf.EMAIL_INTENT_EDITOR_ROLE_NOTIFICATION)
+            self.assertEqual(
+                sent_email_model.subject,
+                self.email_subject_tempate % self.EDITOR_USERNAME)
+
+    def test_correct_rights_are_written_in_manager_role_email_body(self):
+        with self.can_send_emails_ctx, self.can_send_editor_role_email_ctx:
+            # Check that correct email content is sent for Manager.
+            email_manager.send_role_notification_email(
+                self.editor_id, self.new_user_id, rights_manager.ROLE_OWNER,
+                self.exploration.id, self.exploration.title)
+
+            messages = self.mail_stub.get_sent_messages(to=self.NEW_USER_EMAIL)
+            self.assertEqual(len(messages), 1)
+
+            all_models = email_models.SentEmailModel.get_all().fetch()
+            self.assertEqual(len(all_models), 1)
+
+            sent_email_model_manager = all_models[0]
+            self.assertEqual(
```

Comment:
Here and in the next couple of tests: check the plaintext body, and the html body, against the expected strings.